internal/testrunner/runners/asset/runner.go

```diff
@@ -94,60 +104,33 @@ func (r *runner) run(ctx context.Context) ([]testrunner.TestResult, error) {
 	}
 
 	logger.Debug("installing package...")
-	packageInstaller, err := installer.NewForPackage(installer.Options{
-		Kibana:         r.kibanaClient,
-		RootPath:       r.packageRootPath,
-		SkipValidation: true,
-	})
-	if err != nil {
-		return result.WithError(fmt.Errorf("can't create the package installer: %w", err))
-	}
-
-	removePackageHandler := func(ctx context.Context) error {
-		pkgManifest, err := packages.ReadPackageManifestFromPackageRoot(r.packageRootPath)
-		if err != nil {
-			return fmt.Errorf("reading package manifest failed: %w", err)
-		}
-
-		kibanaVersion, err := r.kibanaClient.Version()
-		if err != nil {
-			return fmt.Errorf("failed to retrieve kibana version: %w", err)
-		}
-		stackVersion, err := semver.NewVersion(kibanaVersion.Version())
-		if err != nil {
-			return fmt.Errorf("failed to parse kibana version: %w", err)
-		}
-
-		if stackVersion.LessThan(semver.MustParse("8.0.0")) && pkgManifest.Name == "system" {
-			// in Elastic stack 7.* , system package is installed in the default Agent policy and it cannot be deleted
-			// error: system is installed by default and cannot be removed
-			logger.Debugf("skip uninstalling %s package", pkgManifest.Name)
-			return nil
-		}
-
-		logger.Debug("removing package...")
-		err = packageInstaller.Uninstall(ctx)
-		if err != nil {
-			// logging the error as a warning and not returning it since there could be other reasons that could make fail this process
-			// for instance being defined a test agent policy where this package is used for debugging purposes
-			logger.Warnf("failed to uninstall package %q: %s", pkgManifest.Name, err.Error())
-		}
-		return nil
-	}
-
-	installedPackage, err := packageInstaller.Install(ctx)
+	applyResult, err := r.resourcesManager.ApplyCtx(ctx, r.resources(true))
+	// FIXME: Apply doesn't wrap context errors.
 	if errors.Is(err, context.Canceled) {
 		// Installation interrupted, at this point the package may have been installed, try to remove it for cleanup.
-		err := removePackageHandler(context.WithoutCancel(ctx))
+		_, err := r.resourcesManager.ApplyCtx(context.WithoutCancel(ctx), r.resources(false))
 		if err != nil {
 			logger.Debugf("error while removing package after installation interrupted: %s", err)
 		}
 	}
 	if err != nil {
+		for _, result := range applyResult {
+			if result.Err() != nil {
+				logger.Debugf(result.Err().Error())
+			}
+		}
 		return result.WithError(fmt.Errorf("can't install the package: %w", err))
 	}
 
-	r.removePackageHandler = removePackageHandler
+	manifest, err := packages.ReadPackageManifestFromPackageRoot(r.packageRootPath)
+	if err != nil {
+		return result.WithError(fmt.Errorf("cannot read the package manifest from %s: %w", r.packageRootPath, err))
+	}
+	installedPackage, err := r.kibanaClient.GetPackage(ctx, manifest.Name)
+	if err != nil {
+		return result.WithError(fmt.Errorf("cannot get installed package %q", manifest.Name, err))
+	}
+	installedAssets := installedPackage.Assets()
```

Comment:
On apply we were already getting the assets. We wouldn't need this if `go-resource` had support to get info from defined resources.